nebd/nebd-package/usr/bin/nebd-daemon

```diff
@@ -10,261 +10,305 @@
-    then
-        echo "Can't Write or Create console Log: ${consoleLog}"
-        exit 1
+    PID_FILE=${BASE_LOG_PATH}/nebd-server.pid
+    DAEMON_LOG=${BASE_LOG_PATH}/nebd-server-daemon.log
+    CONSOLE_LOG=${BASE_LOG_PATH}/nebd-server-console.log
+    LOG_PATH=${BASE_LOG_PATH}/server
+    NEBD_CLIENT_LOG_PATH=${BASE_LOG_PATH}/client
+
+    # check config file
+    if [ ! -f ${CONF_PATH} ]; then
+        PREPARE_RES="not found config file '${CONF_PATH}'"
+        # echo "not found config file '${CONF_PATH}'"
+        return
     fi
 
-    # 检查daemonLog是否可写或者是否能够创建
-    touch ${daemonLog} > /dev/null 2>&1
-    if [ $? -ne 0 ]
-    then
-        echo "Can't Write or Create daemon logfile: ${daemonLog}"
-        exit 1
+    create_dir_if_not_exists ${META_DATA_PATH} ${1}
+    create_dir_if_not_exists ${DATA_LOCK_PATH} ${1}
+    create_dir_if_not_exists ${LOG_PATH} ${1}
+    create_dir_if_not_exists ${NEBD_CLIENT_LOG_PATH} ${1}
+}
+
+function start_one() {
+    prepare_env $1
+
+    if [ "${PREPARE_RES}" != "" ]; then
+        echo "$1: prepare env failed, ${PREPARE_RES}"
+        return
     fi
 
-    # 检查jemalloc库文件
-    jemallocpath=`find /usr/ -name "libjemalloc.so*" -print -quit`
-    if [ "${jemallocpath}" = "" ]
-    then
-        echo "Not found jemalloc library in /usr/"
-        exit 1
+    # check whether already started by daemon
+    daemon --name ${DAEMON_NAME} --pidfile ${PID_FILE} --running
+    if [ $? -eq 0 ]; then
+        echo "$1: already started nebd-server by daemon"
+        return
     fi
 
-    # 记录运行daemon前的daemonLog的行数
-    line1=`cat ${daemonLog} | wc -l`
+    # record number of line before start
+    line1=$(cat ${DAEMON_LOG} | wc -l)
+
+    USEROPTS=
+    if $STARTED_AS_ROOT; then
+        USEROPTS="--user ${1}"
+    fi
 
-    LD_PRELOAD=${jemallocpath} daemon --name nebd-server --core --inherit \
-           --respawn --attempts 10 --delay 10 --acceptable 10\
-           --pidfile ${pidFile} \
-           --errlog ${daemonLog} \
-           --output ${consoleLog} \
-           -- ${bin} -confPath=${confPath} -log_dir=${logPath} -graceful_quit_on_sigterm=true -stderrthreshold=3
+    LD_PRELOAD=${jemallocpath} daemon --name ${DAEMON_NAME} --core --inherit \
+        --respawn --attempts 10 --delay 10 --acceptable 10 --pidfile ${PID_FILE} \
+        --errlog ${DAEMON_LOG} \
+        --output ${CONSOLE_LOG} ${USEROPTS} \
+        -- ${NEBD_SERVER_BIN} -confPath=${CONF_PATH} -log_dir=${LOG_PATH} -graceful_quit_on_sigterm=true -stderrthreshold=3
 
-    # sleep 1秒检测daemonLog是否有变化，如果有变化，说明启动遇到问题
+    # sleep a while, check whether start successful
     sleep 1
-    line2=`cat ${daemonLog} | wc -l`
-    if [ $line1 != $line2 ]
-    then
-        echo "start nebd-server met error!"
-        stop
+    line2=$(cat ${DAEMON_LOG} | wc -l)
+    if [ $line1 != $line2 ]; then
+        echo "$1: start nebd-server met error!"
+        stop_one $1
     fi
 }
 
-# 停止daemon进程，且停止nebd-server
-function stop() {
+function stop_one() {
```

Comment:
An administrator or maintener can only  start  most two nebd-server(user of root and himself) ?

Reply:
if start/stop/restart/status executed by root or with sudo, it will execute for every users that recoreded in `/etc/nebd/users` and root user which it default included.
operations with `_one` suffix only for current user, event run by root or with sudo.